src/graph_gui.cpp

```diff
@@ -1329,6 +1332,267 @@ struct PerformanceRatingDetailWindow : Window {
+					if (row-- > 0) continue;
+
+					cs = CargoSpec::Get(i->produced_cargo[j]);
+
+					ToggleBit(_legend_excluded_cargo, cs->Index());
+					this->UpdateExcludedData();
+					this->SetDirty();
+					break;
+				}
+				break;
+			}
+		}
+	}
+
+	void SetStringParameters(int widget) const override
+	{
+		if (widget == WID_IP_CAPTION) SetDParam(0, this->window_number);
+	}
+
+	void OnResize() override
+	{
+		this->vscroll->SetCapacityFromWidget(this, WID_IP_MATRIX);
+	}
+
+	void UpdateStatistics(bool initialize) override
+	{
+		CargoTypes excluded_cargo = this->excluded_data;
+		this->UpdateExcludedData();
+
+		int mo = _cur_month - this->num_vert_lines;
+		int yr = _cur_year;
+		while (mo < 0) {
+			yr--;
+			mo += 12;
+		}
+
+		if (!initialize && this->excluded_data == excluded_cargo && this->num_on_x_axis == this->num_vert_lines && this->year == yr && this->month == mo) {
+			/* There's no reason to get new stats */
+			return;
+		}
+
+		this->year = yr;
+		this->month = mo;
+
+		int index = 0;
+		const Industry *i = Industry::Get(this->window_number);
+		for (size_t k = 0; k < INDUSTRY_NUM_OUTPUTS; k++) {
+			if (!IsCargoIDValid(i->produced_cargo[k])) continue;
+
+			const CargoSpec *cs = CargoSpec::Get(i->produced_cargo[k]);
+
+			this->colours[index] = cs->legend_colour;
+			for (uint j = 0; j < GRAPH_NUM_MONTHS; j++) {
+				this->cost[index][j] = Industry::Get(this->window_number)->past_production[GRAPH_NUM_MONTHS-j-1][k];
+			}
+			index++;
+		}
+		this->num_dataset = index;
+		this->vscroll->SetCount(index);
+
+		this->SetDirty();
+	}
+};
+
+static const NWidgetPart _nested_industry_production_widgets[] = {
+	NWidget(NWID_HORIZONTAL),
+		NWidget(WWT_CLOSEBOX, COLOUR_BROWN),
+		NWidget(WWT_CAPTION, COLOUR_BROWN, WID_IP_CAPTION), SetDataTip(STR_GRAPH_INDUSTRY_PRODUCTION_CAPTION, STR_TOOLTIP_WINDOW_TITLE_DRAG_THIS),
+		NWidget(WWT_SHADEBOX, COLOUR_BROWN),
+		NWidget(WWT_DEFSIZEBOX, COLOUR_BROWN),
+		NWidget(WWT_STICKYBOX, COLOUR_BROWN),
+	EndContainer(),
+	NWidget(WWT_PANEL, COLOUR_BROWN, WID_IP_BACKGROUND), SetMinimalSize(568, 128),
+		NWidget(NWID_HORIZONTAL),
+			NWidget(WWT_EMPTY, COLOUR_BROWN, WID_IP_GRAPH), SetMinimalSize(495, 0), SetFill(1, 1), SetResize(1, 1),
+			NWidget(NWID_VERTICAL),
+				NWidget(NWID_SPACER), SetMinimalSize(0, 24), SetFill(0, 1),
+				NWidget(WWT_PUSHTXTBTN, COLOUR_ORANGE, WID_IP_ENABLE_CARGOES), SetDataTip(STR_GRAPH_CARGO_ENABLE_ALL, STR_GRAPH_CARGO_TOOLTIP_ENABLE_ALL), SetFill(1, 0),
+				NWidget(WWT_PUSHTXTBTN, COLOUR_ORANGE, WID_IP_DISABLE_CARGOES), SetDataTip(STR_GRAPH_CARGO_DISABLE_ALL, STR_GRAPH_CARGO_TOOLTIP_DISABLE_ALL), SetFill(1, 0),
+				NWidget(NWID_SPACER), SetMinimalSize(0, 4),
+				NWidget(NWID_HORIZONTAL),
+					NWidget(WWT_MATRIX, COLOUR_ORANGE, WID_IP_MATRIX), SetResize(0, 2), SetMatrixDataTip(1, 0, STR_GRAPH_CARGO_PAYMENT_TOGGLE_CARGO), SetScrollbar(WID_IP_MATRIX_SCROLLBAR),
+					NWidget(NWID_VSCROLLBAR, COLOUR_ORANGE, WID_IP_MATRIX_SCROLLBAR),
```

Comment:
In the similar Cargo Payment Rates GUI, these are all COLOUR_BROWN to match the window background.